docs/bigquery-usage.rst

```diff
@@ -291,26 +291,88 @@ Run a query which can be expected to complete within bounded time:
 
    >>> from gcloud import bigquery
    >>> client = bigquery.Client()
-   >>> query = """\
+   >>> QUERY = """\
    SELECT count(*) AS age_count FROM dataset_name.person_ages
    """
-   >>> query = client.run_sync_query(query)
+   >>> query = client.run_sync_query(QUERY)
    >>> query.timeout_ms = 1000
    >>> query.run()  # API request
+   >>> query.complete
+   True
+   >>> len(query.schema)
+   0
+   >>> field = query.schema[0]
+   >>> field.name
+   u'count'
+   >>> field.field_type
+   u'INTEGER'
+   >>> field.mode
+   u'NULLABLE'
+   >>> query.rows
+   [(15,)]
+   >>> query.total_rows
+   1
+
+If the rows returned by the query do not fit into the inital response,
+then we need to fetch the remaining rows via ``fetch_data``:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> QUERY = """\
+   SELECT * FROM dataset_name.person_ages
+   """
+   >>> query = client.run_sync_query(QUERY)
+   >>> query.timeout_ms = 1000
+   >>> query.run()  # API request
+   >>> query.complete
+   True
+   >>> query.total_rows
+   1234
+   >>> query.page_token
+   'TOKEN'
+   >>> do_something_with(query.schema, query.rows)
+   >>> token = query_page_token  # for initial request
+   >>> while true:
```

Comment:
I disagree:  this is a "do ... while" loop, for which `while True:`  with a test-and-break is the idiomatic Python spelling (see the rejected [PEP 315](https://www.python.org/dev/peps/pep-0315/)).  It is a different case than a "retry/backoff", where a counter would be needed.